src/charm.py

```diff
@@ -220,6 +235,20 @@ def _on_list_receivers_action(self, event: ops.ActionEvent):
             res[receiver.replace("_", "-")] = self.get_receiver_url(receiver)
         event.set_results(res)
 
+    def _on_remote_write_changed(self, _event):
+        """Event handler for the remote write changed event."""
+        # notify the cluster
+        self.coordinator.update_cluster()
+
+    def _on_collect_status(self, e: CollectStatusEvent):
+        # add Tempo coordinator-specific statuses
+        if not self.remote_write_endpoints():
+            e.add_status(
+                ops.ActiveStatus(
+                    "metrics-generator disabled. Add a relation over send-remote-write"
```

Comment:
this is only relevant if we have a worker node with metrics-generator enabled.
IMHO we should be doing something like:
```
if 'metrics-generator' in self.coordinator.worker_roles and not self.remote_write_endpoints():
    ... set active
```